{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"http://lssds.aura-astronomy.org/winter_school/sites/default/files/sods_atfdds_header01.jpg\" alt=\"La Serena School for Data Science: Applied Tools for Data-driven Sciences\" />\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Introducción:\n",
    "\n",
    "Un Dockerfile es un archivo de configuración en formato de texto plano utilizado para definir los pasos para construir una imagen de Docker. Es un componente crítico del ecosistema de Docker, que permite a los desarrolladores automatizar el proceso de creación de imágenes de contenedor reproducibles y consistentes. Los Dockerfiles se utilizan para encapsular la configuración y dependencias de una aplicación o servicio, lo que facilita empaquetar, distribuir e implementar software en forma de contenedores.\n",
    "\n",
    "El Dockerfile incluye típicamente una serie de instrucciones, cada una en una línea separada, que dictan cómo debe construirse la imagen. El motor de Docker interpreta estas instrucciones y construye la imagen capa por capa. Cuando se crea un nuevo contenedor a partir de la imagen, el sistema de archivos se construye en base a estas capas, lo que lo hace eficiente y rápido.\n",
    "\n",
    "A continuación se muestra un ejemplo de un Dockerfile para crear un servidor de Jupyter Notebook utilizando la imagen oficial de Jupyter Docker Stacks:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FROM jupyter/base-notebook:latest\n",
      "\n",
      "WORKDIR /home/jovyan/work\n",
      "\n",
      "COPY . /home/jovyan/work\n",
      "\n",
      "EXPOSE 8889\n",
      "\n",
      "RUN pip install pandas matplotlib\n",
      "\n",
      "CMD [\"jupyter\", \"notebook\", \"--ip=0.0.0.0\", \"--port=8889\",\"--no-browser\"]\n"
     ]
    }
   ],
   "source": [
    "!cat Dockerfile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Componentes clave de un Dockerfile:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- **2.1 Imagen Base:**\n",
    "\n",
    "El Dockerfile comienza especificando la imagen base sobre la cual se construirá la imagen personalizada. La imagen base es el punto de partida para la imagen de tu aplicación y contiene el sistema operativo y otro software o bibliotecas preinstaladas.\n",
    "La imagen base se define utilizando la instrucción FROM. Por ejemplo:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "FROM jupyter/minimal-notebook"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- **2.2 Configuración del Entorno:**\n",
    "\n",
    "El Dockerfile puede incluir configuraciones del entorno que definan variables, especifiquen directorios de trabajo y establezcan otros comportamientos en tiempo de ejecución.\n",
    "La configuración del entorno se establece utilizando la instrucción ENV. Por ejemplo:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ENV NODE_ENV=production"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- **2.3 Copiar Archivos:**\n",
    "\n",
    "La instrucción COPY te permite copiar archivos y directorios desde el sistema anfitrión al sistema de archivos de la imagen. Este paso es esencial para agregar el código de tu aplicación y cualquier recurso necesario a la imagen.\n",
    "Por ejemplo:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "COPY . /app"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- **2.4 Ejecución de Comandos:**\n",
    "\n",
    "La instrucción RUN se utiliza para ejecutar comandos dentro del sistema de archivos de la imagen. Esto se utiliza comúnmente para instalar paquetes, dependencias y otro software necesario para que tu aplicación funcione.\n",
    "\n",
    "Por ejemplo, para una aplicación Node.js, podrías ejecutar el siguiente comando para instalar las dependencias de Node.js:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "RUN npm install"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- **2.5 Exposición de Puertos:**\n",
    "\n",
    "La instrucción EXPOSE se utiliza para especificar en qué puertos debe escuchar el contenedor. No publica los puertos en realidad, sino que sirve como documentación para los desarrolladores y usuarios de la imagen.\n",
    "\n",
    "Por ejemplo, para indicar que el contenedor escucha en el puerto 3000, agregarías la siguiente línea:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "EXPOSE 3000"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- **2.6 Comando de Ejecución del Contenedor:**\n",
    "\n",
    "La instrucción ***CMD*** o ***ENTRYPOINT*** define el comando que se debe ejecutar cuando se inicia un contenedor a partir de la imagen. Por lo general, este es el proceso principal de tu aplicación.\n",
    "\n",
    "Por ejemplo, para una aplicación Node.js, podrías usar:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "CMD [\"node\", \"app.js\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Una vez que hayas definido tu Dockerfile, puedes usar el comando ***docker build*** para construir la imagen de Docker basada en las instrucciones proporcionadas en el archivo. Esto creará una imagen de contenedor que contiene todas las configuraciones, dependencias y código especificados necesarios para ejecutar tu aplicación.\n",
    "\n",
    "Los Dockerfiles son una herramienta poderosa para crear imágenes de contenedor consistentes y reproducibles. Permiten a los desarrolladores definir el entorno exacto necesario para sus aplicaciones, lo que facilita compartir y distribuir el software en diferentes entornos y sistemas."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Crear un Dockerfile paso a paso:\n",
    "\n",
    "Vamos a crear un Dockerfile paso a paso para un servidor de Jupyter Notebook utilizando la imagen oficial de Jupyter Docker Stacks:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- **3.1: Elegir la Imagen Base**\n",
    "\n",
    "Utilizaremos la imagen oficial de Jupyter Notebook de Jupyter Docker Stacks como imagen base. Esta imagen incluye la instalación básica de Jupyter Notebook.\n",
    "Elige la etiqueta (tag) de la imagen base según tus requisitos. En este ejemplo, utilizaremos la etiqueta ***base-notebook***."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "FROM jupyter/base-notebook:latest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- **3.2: Establecer el Directorio de Trabajo**\n",
    "\n",
    "Es una buena práctica establecer el directorio de trabajo dentro del contenedor donde se ubicarán tus archivos de Jupyter Notebook.\n",
    "Lo estableceremos en ***/home/jovyan/work***, que es el directorio predeterminado para los archivos de Jupyter Notebook en la imagen base."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "WORKDIR /home/jovyan/work"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- **3.3: Copiar Archivos de Notebooks**\n",
    "\n",
    "A continuación, copiaremos los archivos de Jupyter Notebook desde el directorio local (donde se encuentra el Dockerfile) al directorio de trabajo del contenedor.\n",
    "Asegúrate de que tus archivos de Jupyter Notebook (por ejemplo, archivos ***.ipynb***) estén presentes en el mismo directorio que el Dockerfile."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "COPY . /home/jovyan/work"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- **3.4: Exponer Puerto**\n",
    "\n",
    "El Jupyter Notebook se ejecuta de forma predeterminada en el puerto 8888, pero puede haber casos en los que se use este puerto, por lo que lo cambiaremos a 8889. Para acceder a él desde la máquina anfitriona, necesitamos exponer este puerto desde el contenedor."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "EXPOSE 8889"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- **3.5: Instalar Paquetes Adicionales (Opcional)**\n",
    "\n",
    "Si tus Jupyter Notebooks requieren paquetes o bibliotecas adicionales de Python, puedes instalarlos utilizando pip o conda.\n",
    "En este ejemplo, utilizaremos pip para instalar pandas y matplotlib."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "RUN pip install pandas matplotlib"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- **3.6: Iniciar Servidor de Jupyter Notebook**\n",
    "\n",
    "Finalmente, agregaremos el comando para iniciar el servidor de Jupyter Notebook cuando se ejecute el contenedor.\n",
    "La opción --ip=0.0.0.0 permite el acceso al servidor desde todas las direcciones IP, y --no-browser evita que Jupyter intente abrir un navegador dentro del contenedor."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "CMD [\"jupyter\", \"notebook\", \"--ip=0.0.0.0\", \"--port=8889\",\"--no-browser\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "El Dockerfile completo se verá así:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FROM jupyter/base-notebook:latest\n",
      "\n",
      "WORKDIR /home/jovyan/work\n",
      "\n",
      "COPY . /home/jovyan/work\n",
      "\n",
      "EXPOSE 8889\n",
      "\n",
      "RUN pip install pandas matplotlib\n",
      "\n",
      "CMD [\"jupyter\", \"notebook\", \"--ip=0.0.0.0\", \"--port=8889\",\"--no-browser\"]\n"
     ]
    }
   ],
   "source": [
    "!cat Dockerfile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To build the Docker image, save the above code into a file named Dockerfile in the same directory as your Jupyter Notebook files. Then, open a terminal or command prompt, navigate to the directory containing the Dockerfile, and run the following command:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[1A\u001b[1B\u001b[0G\u001b[?25l[+] Building 0.0s (0/1)                                                         \n",
      "\u001b[?25h\u001b[1A\u001b[0G\u001b[?25l[+] Building 0.1s (2/3)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 0.3s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                0.0s\n",
      "\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 0.4s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                0.2s\n",
      "\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 0.6s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                0.4s\n",
      "\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 0.7s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                0.5s\n",
      "\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 0.9s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                0.7s\n",
      "\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 1.0s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                0.8s\n",
      "\u001b[2m => => # WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508444210>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 1.2s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                1.0s\n",
      "\u001b[2m => => # WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508444210>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 1.4s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                1.1s\n",
      "\u001b[2m => => # WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508444210>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 1.5s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                1.3s\n",
      "\u001b[2m => => # WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508444210>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 1.7s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                1.4s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508444d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 1.8s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                1.6s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508444d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 2.0s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                1.7s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508444d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 2.1s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                1.9s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508444d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 2.3s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                2.0s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508444d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 2.4s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                2.2s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508444d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 2.5s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                2.3s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 2.7s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                2.4s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 2.8s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                2.6s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 3.0s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                2.7s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 3.1s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                2.9s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 3.3s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                3.0s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 3.4s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                3.2s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 3.6s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                3.3s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 3.7s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                3.5s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 3.9s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                3.6s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 4.0s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                3.8s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 4.2s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                3.9s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 4.3s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                4.1s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 4.5s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                4.2s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 4.6s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                4.4s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 4.8s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                4.5s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 4.9s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                4.7s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 5.1s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                4.8s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 5.2s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                5.0s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 5.4s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                5.1s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 5.5s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                5.3s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 5.7s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                5.4s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 5.8s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                5.6s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 6.0s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                5.7s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 6.1s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                5.9s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 6.3s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                6.0s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 6.4s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                6.2s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 6.6s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                6.3s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 6.7s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                6.5s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 6.9s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                6.6s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 7.0s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                6.8s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 7.2s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                6.9s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 7.3s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                7.1s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 7.5s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                7.2s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 7.6s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                7.4s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 7.8s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                7.5s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 7.9s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                7.7s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 8.1s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                7.8s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 8.2s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                8.0s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 8.4s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                8.1s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 8.5s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                8.3s\n",
      "\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=No\n",
      "\u001b[0m\u001b[2m => => # ne, status=None)) after connection broken by 'NewConnectionError('<pip\n",
      "\u001b[0m\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508446790>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 8.7s (7/8)                                                         \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m => [4/4] RUN pip install pandas matplotlib                                8.4s\n",
      "\u001b[2m => => # ._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508446790>:\n",
      "\u001b[0m\u001b[2m => => #  Failed to establish a new connection: [Errno -2] Name or service not \n",
      "\u001b[0m\u001b[2m => => # known')': /simple/pandas/                                             \n",
      "\u001b[0m\u001b[2m => => # ERROR: Could not find a version that satisfies the requirement pandas \n",
      "\u001b[0m\u001b[2m => => # (from versions: none)                                                 \n",
      "\u001b[0m\u001b[2m => => # ERROR: No matching distribution found for pandas                      \n",
      "\u001b[0m\u001b[?25h\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[1A\u001b[0G\u001b[?25l[+] Building 8.7s (8/8) FINISHED                                                \n",
      "\u001b[34m => [internal] load build definition from Dockerfile                       0.0s\n",
      "\u001b[0m\u001b[34m => => transferring dockerfile: 37B                                        0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load .dockerignore                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 2B                                            0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load metadata for docker.io/jupyter/base-notebook:latest    0.0s\n",
      "\u001b[0m\u001b[34m => [1/4] FROM docker.io/jupyter/base-notebook:latest                      0.0s\n",
      "\u001b[0m\u001b[34m => [internal] load build context                                          0.0s\n",
      "\u001b[0m\u001b[34m => => transferring context: 558.53kB                                      0.0s\n",
      "\u001b[0m\u001b[34m => CACHED [2/4] WORKDIR /home/jovyan/work                                 0.0s\n",
      "\u001b[0m\u001b[34m => [3/4] COPY . /home/jovyan/work                                         0.1s\n",
      "\u001b[0m\u001b[31m => ERROR [4/4] RUN pip install pandas matplotlib                          8.5s\n",
      "\u001b[0m                                                                                \n",
      "                                                                                \n",
      "                                                                                \n",
      "                                                                                \n",
      "                                                                                \n",
      "                                                                                \n",
      "\u001b[6A\u001b[0G\u001b[?25h------\n",
      " > [4/4] RUN pip install pandas matplotlib:\n",
      "#8 0.754 WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508444210>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/pandas/\n",
      "#8 1.260 WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508444d90>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/pandas/\n",
      "#8 2.271 WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445450>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/pandas/\n",
      "#8 4.278 WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508445d90>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/pandas/\n",
      "#8 8.287 WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f4508446790>: Failed to establish a new connection: [Errno -2] Name or service not known')': /simple/pandas/\n",
      "#8 8.305 ERROR: Could not find a version that satisfies the requirement pandas (from versions: none)\n",
      "#8 8.305 ERROR: No matching distribution found for pandas\n",
      "------\n",
      "executor failed running [/bin/bash -o pipefail -c pip install pandas matplotlib]: exit code: 1\n"
     ]
    }
   ],
   "source": [
    "!docker build -t jupyter_server ."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The build process will create a Docker image with your Jupyter Notebook server configuration.\n",
    "\n",
    "To run the Jupyter Notebook server in a container based on the image you just built, use the following command:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[I 2023-08-12 19:19:48.106 ServerApp] Package notebook took 0.0000s to import\n",
      "[I 2023-08-12 19:19:48.131 ServerApp] Package jupyter_lsp took 0.0172s to import\n",
      "[W 2023-08-12 19:19:48.131 ServerApp] A `_jupyter_server_extension_points` function was not found in jupyter_lsp. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.\n",
      "[I 2023-08-12 19:19:48.140 ServerApp] Package jupyter_server_terminals took 0.0086s to import\n",
      "[I 2023-08-12 19:19:48.141 ServerApp] Package jupyterlab took 0.0000s to import\n",
      "[I 2023-08-12 19:19:48.705 ServerApp] Package nbclassic took 0.0019s to import\n",
      "[W 2023-08-12 19:19:48.707 ServerApp] A `_jupyter_server_extension_points` function was not found in nbclassic. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.\n",
      "[I 2023-08-12 19:19:48.707 ServerApp] Package notebook_shim took 0.0000s to import\n",
      "[W 2023-08-12 19:19:48.707 ServerApp] A `_jupyter_server_extension_points` function was not found in notebook_shim. Instead, a `_jupyter_server_extension_paths` function was found and will be used for now. This function name will be deprecated in future releases of Jupyter Server.\n",
      "[I 2023-08-12 19:19:48.708 ServerApp] jupyter_lsp | extension was successfully linked.\n",
      "[I 2023-08-12 19:19:48.712 ServerApp] jupyter_server_terminals | extension was successfully linked.\n",
      "[I 2023-08-12 19:19:48.716 ServerApp] jupyterlab | extension was successfully linked.\n",
      "[I 2023-08-12 19:19:48.720 ServerApp] nbclassic | extension was successfully linked.\n",
      "[I 2023-08-12 19:19:48.724 ServerApp] notebook | extension was successfully linked.\n",
      "[I 2023-08-12 19:19:48.725 ServerApp] Writing Jupyter server cookie secret to /home/jovyan/.local/share/jupyter/runtime/jupyter_cookie_secret\n",
      "[I 2023-08-12 19:19:48.943 ServerApp] notebook_shim | extension was successfully linked.\n",
      "[I 2023-08-12 19:19:48.971 ServerApp] notebook_shim | extension was successfully loaded.\n",
      "[I 2023-08-12 19:19:48.973 ServerApp] jupyter_lsp | extension was successfully loaded.\n",
      "[I 2023-08-12 19:19:48.974 ServerApp] jupyter_server_terminals | extension was successfully loaded.\n",
      "[I 2023-08-12 19:19:48.976 LabApp] JupyterLab extension loaded from /opt/conda/lib/python3.11/site-packages/jupyterlab\n",
      "[I 2023-08-12 19:19:48.976 LabApp] JupyterLab application directory is /opt/conda/share/jupyter/lab\n",
      "[I 2023-08-12 19:19:48.977 LabApp] Extension Manager is 'pypi'.\n",
      "[I 2023-08-12 19:19:48.980 ServerApp] jupyterlab | extension was successfully loaded.\n",
      "[I 2023-08-12 19:19:48.983 ServerApp] nbclassic | extension was successfully loaded.\n",
      "[I 2023-08-12 19:19:48.986 ServerApp] notebook | extension was successfully loaded.\n",
      "[I 2023-08-12 19:19:48.986 ServerApp] Serving notebooks from local directory: /home/jovyan/work\n",
      "[I 2023-08-12 19:19:48.987 ServerApp] Jupyter Server 2.7.0 is running at:\n",
      "[I 2023-08-12 19:19:48.987 ServerApp] http://6786521305f5:8889/tree?token=bf10af5a3f4bd15a0d617b26c87563066f6683d495949e4e\n",
      "[I 2023-08-12 19:19:48.987 ServerApp]     http://127.0.0.1:8889/tree?token=bf10af5a3f4bd15a0d617b26c87563066f6683d495949e4e\n",
      "[I 2023-08-12 19:19:48.987 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).\n",
      "[C 2023-08-12 19:19:48.990 ServerApp] \n",
      "    \n",
      "    To access the server, open this file in a browser:\n",
      "        file:///home/jovyan/.local/share/jupyter/runtime/jpserver-7-open.html\n",
      "    Or copy and paste one of these URLs:\n",
      "        http://6786521305f5:8889/tree?token=bf10af5a3f4bd15a0d617b26c87563066f6683d495949e4e\n",
      "        http://127.0.0.1:8889/tree?token=bf10af5a3f4bd15a0d617b26c87563066f6683d495949e4e\n",
      "[I 2023-08-12 19:19:49.564 ServerApp] Skipped non-installed server(s): bash-language-server, dockerfile-language-server-nodejs, javascript-typescript-langserver, jedi-language-server, julia-language-server, pyright, python-language-server, python-lsp-server, r-languageserver, sql-language-server, texlab, typescript-language-server, unified-language-server, vscode-css-languageserver-bin, vscode-html-languageserver-bin, vscode-json-languageserver-bin, yaml-language-server\n",
      "0.00s - Debugger warning: It seems that frozen modules are being used, which may\n",
      "0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off\n",
      "0.00s - to python to disable frozen modules.\n",
      "0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.\n",
      "^C\n",
      "\n",
      "  _   _          _      _\n",
      " | | | |_ __  __| |__ _| |_ ___\n",
      " | |_| | '_ \\/ _` / _` |  _/ -_)\n",
      "  \\___/| .__/\\__,_\\__,_|\\__\\___|\n",
      "       |_|\n",
      "                                                                           \n",
      "Read the migration plan to Notebook 7 to learn about the new features and the actions to take if you are using extensions.\n",
      "\n",
      "https://jupyter-notebook.readthedocs.io/en/latest/migrate_to_notebook7.html\n",
      "\n",
      "Please note that updating to Notebook 7 might break some of your extensions.\n",
      "\n",
      "[I 2023-08-12 19:22:22.457 ServerApp] Interrupted...\n"
     ]
    }
   ],
   "source": [
    "!docker run -p 8889:8889 jupyter_server"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "El **-p 8889:8889** mapea el puerto 8889 del contenedor al puerto 8889 de la máquina anfitriona. Luego, puedes acceder al servidor de Jupyter Notebook abriendo un navegador web y navegando a http://localhost:8889 en la máquina anfitriona. Verás la interfaz de Jupyter Notebook y podrás comenzar a trabajar con tus notebooks."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
